storybookjs/storybook
# Viewport styles

## What I did

Added a small bit of code to the addon/viewport that adds a data attribute to the `iframe body` element (containing `#storybook-root`) called `data-storybook-viewport-id`. This lets you define viewport specific styles in `.storybook/preview.css` such as 

```css
#storybook-root {
  padding: 12px;
}
body[data-storybook-viewport-id="mobile1"] #storybook-root {
  padding: 8px;
}
body[data-storybook-viewport-id="mobile2"] #storybook-root {
  padding: 16px;
}
```

Currently only the height and width of the viewport can be adjusted, but with this change if you want to add other attributes you can define them in CSS without changing the API.

## Checklist for Contributors

### Testing



#### The changes in this PR are covered in the following automated tests:

#### Manual testing

1. Run `yarn storybook:ui` in `code\`
2. Open Storybook in your browser
3. Select a component to preview (such as "Example\Button")
4. Open the Developer Console run the command: ```javascript
document.querySelector('iframe[data-is-storybook="true"]').contentDocument.querySelector('body').dataset.storybookViewportId
```
5. This should print the selected viewport's id (If no viewport is selected it should print the word 'reset'.)
6. Change the viewport and rerun the JS command to see the change.

### Documentation



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release



This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_
